shared/js/background/ready.es6.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Module that exports an API for announcing that, or checking if, a part of the
+ * extension's code is ready for use. This helps to avoid circular dependencies
+ * that can otherwise result from exporting ready functions directly from each
+ * module.
+ */
+
+/**
+ * @typedef readyResolverResult
```

Comment:
Can we make this a class:
```suggestion
class ReadyResolverResult {
    constructor() {
        this._readyResolver = null
        this._readyPromise = new Promise(resolve => { _readyResolver = resolve })
    }
    ready () { return this._readyPromise },
    resolve () {
        if (this._readyResolver) {
            this._readyResolver()
            this._readyResolver = null
        }
    }
}

export const startup = new ReadyResolverResult()
```

Reply:
Sure if you prefer, done.